WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java

```diff
@@ -0,0 +1,402 @@
+package org.wordpress.android.push;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.IBinder;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.NotificationManagerCompat;
+import android.support.v4.app.RemoteInput;
+import android.text.TextUtils;
+
+import com.android.volley.VolleyError;
+import com.wordpress.rest.RestRequest;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.analytics.AnalyticsTracker;
+import org.wordpress.android.models.CommentStatus;
+import org.wordpress.android.models.Note;
+import org.wordpress.android.ui.comments.CommentActionResult;
+import org.wordpress.android.ui.comments.CommentActions;
+import org.wordpress.android.ui.main.WPMainActivity;
+import org.wordpress.android.ui.notifications.NotificationsListFragment;
+import org.wordpress.android.util.AppLog;
+
+import java.util.HashMap;
+
+import static org.wordpress.android.push.GCMMessageService.ACTIONS_RESULT_NOTIFICATION_ID;
+import static org.wordpress.android.push.GCMMessageService.EXTRA_VOICE_REPLY;
+import static org.wordpress.android.ui.notifications.NotificationsListFragment.NOTE_INSTANT_REPLY_EXTRA;
+
+/**
+ * service which makes it possible to process Notifications quick actions in the background,
+ * such as:
+ * - like
+ * - reply-to-comment
+ * - approve
+ */
+
+public class NotificationsProcessingService extends Service {
+
+    public static final String ARG_ACTION_TYPE = "action_type";
+    public static final String ARG_ACTION_LIKE = "action_like";
+    public static final String ARG_ACTION_REPLY = "action_reply";
+    public static final String ARG_ACTION_APPROVE = "action_approve";
+    public static final String ARG_ACTION_REPLY_TEXT = "action_reply_text";
+    public static final String ARG_NOTE_ID = "note_id";
+
+    public static void startServiceForLike(Context context, String noteId) {
+        Intent intent = new Intent(context, NotificationsProcessingService.class);
+        intent.putExtra(ARG_ACTION_TYPE, ARG_ACTION_LIKE);
+        intent.putExtra(ARG_NOTE_ID, noteId);
+        context.startService(intent);
+    }
+
+    public static void startServiceForApprove(Context context, String noteId) {
+        Intent intent = new Intent(context, NotificationsProcessingService.class);
+        intent.putExtra(ARG_ACTION_TYPE, ARG_ACTION_APPROVE);
+        intent.putExtra(ARG_NOTE_ID, noteId);
+        context.startService(intent);
+    }
+
+    public static void startServiceForReply(Context context, String noteId, String replyToComment) {
```

Comment:
I should have added more comments there - yes these 3 methods are not used as of now, but I decided to leave them on purpose in case someone wants to run the service, as an interface for a potential user, so they don't need to build the intent with the right parameters by hand. They can just invoke any of these 3 params for having the service handle any of the 3 implemented actions (Reply/Like/Approve)